### ISSUE TITLE
Discord privacy

### ISSUE THREAD

**Issue by DasBambus** [powercord-community/suggestions#721]
### Description
Just so Discord can't read all your messages and can't collect any of your Data

### More info
I had the idea when I watched a youtube video about Discord might collecting your data because the CEO already did it in the past with his old company called openfeint. It was also confirmed that discord can read your private messages.
So this plugin should just stop discord from collecting your data.
YouTube Video: The Disturbing Truth about Discord | by Visual Venture
Link: https://www.youtube.com/watch?v=M5X6iLHzqAI

----

 - [ x] I checked if this request wasn't already proposed
 - [ x] I made sure this request follows Powercord's [plugin guidelines](https://powercord.dev/guidelines)

**Comment by BoredManCodes** [comment]
I am not sure how this could be possible
To send a message you need to use Discord's servers. This means your message passes through them.

About the only thing I can think of is to encrypt your messages using a plugin and have the recipient use the same plugin and public key

**Comment by DasBambus** [comment]
So it's possible but it only works when message someone who is using the plugin as well

**Comment by BoredManCodes** [comment]
Yeah, but it'd always be sent encrypted (otherwise what's the point) so you'd need to assume the other person has the plugin otherwise it's unreadable. At that point you'd be better messaging them using whatsapp or matrix.

I am not a Powercord dev though so I may be missing something

**Comment by colin273** [comment]
Oh, so you mean something like this: https://github.com/SammCheese/invisible-chat

Or this: https://github.com/SammCheese/SimpleDiscordCryptLoader

**Comment by BoredManCodes** [comment]
> I am not a Powercord dev though so I ~~may be~~ am missing something

**Comment by BoredManCodes** [comment]
But it still relies on them having the plugin, and you are likely going to be sharing the encryption key with them via discord anyway :shrug:

**Comment by DasBambus** [comment]
oh so when I share the encryption key over discord the whole purpose of the plugin is destroyed

**Comment by DasBambus** [comment]
and when I share it over whatsapp I can use whatsapp instead

**Comment by BoredManCodes** [comment]
My points exactly :)

**Comment by DasBambus** [comment]
what is matrix I never heard of this

**Comment by BenSegal855** [comment]
If you want encrypted messaging, then use an encrypted messaging service. Discord is not encrypted and properly never will be. There are ways to tack encryption onto discord but, as you mentioned, they can be difficult to setup securely.

**Comment by quincynyan** [comment]
Imagine the plugin using a third party API to send the encrypted message (not through the discord servers) then decrypting and displaying them on the other end with the plugin. Oh wait why don't we just code another messaging app at this point lmao? Ben makes a good point, if you don't like how the discord servers work, use another app. Powercord is *mostly* centralized around discord.

**Comment by DasBambus** [comment]
Yeah I see I just don't know any app with similar options an accessibility.

**Comment by quincynyan** [comment]
Theoretically speaking, you could have a private backend where you would send your actual messages, which will then send to the recipient and displayed in Discord. It's like basically creating another messaging app without a UI and using the Discord UI. I doubt the Powercord team will allow such a plugin, and even if they did, it'd be very ineffective.

**Comment by saint2706** [comment]
this sounds like fosscord

**Comment by Swishilicous** [comment]
this sounds like fosscord

**Comment by spinfal** [comment]
If you don't want Discord to collect most of your data/usage activity, you should turn these off:

![Settings > Privacy & Safety](https://cdn.spin.rip/r/7ec17155-3ae1-4d68-a27e-021dc8318ec5.png "Settings > Privacy & Safety")
*Settings > Privacy & Safety*
<br>
![Settings > Voice & Sound](https://cdn.spin.rip/r/80f8b436-3c68-4133-be79-2cdff88652bd.png "Settings > Voice & Sound")
*Settings > Voice & Sound*

As for preventing Discord from reading your messages? That isn't really possible or it isn't easy at all (like the other comments are stating)

**Comment by Henry-Hiles** [comment]
Also turn off sentry. (There are various ways)

**Comment by Henry-Hiles** [comment]
Also turn off sentry. (There are various ways)

**Comment by adntaha** [comment]
> Also turn off sentry. (There are various ways)

Powercord already does this by default. ([source](https://github.com/powercord-org/powercord/blob/v2/src/Powercord/coremods/no-track/index.js#L18-L21))
